src/main/src/main/java/org/geoserver/catalog/CoverageViewReader.java

```diff
@@ -388,12 +420,106 @@ public GridCoverage2D read(GeneralParameterValue[] parameters) throws IllegalArg
         return result;
     }
 
+    private GridCoverage2D readUsingJiffle(ViewInputs viewInputs) throws IOException {
+        // TODO: Optimize this to introduce a bandSelect and bands remap when dealing
+        // with coverages with many bands.
+        HashMap<String, GridCoverage2D> inputCoverages = viewInputs.getInputCoverages();
+        GridCoverage2D[] coverages = inputCoverages.values().toArray(new GridCoverage2D[0]);
+        GridCoverage2D reference = coverages[0];
+
+        String destName = coverageView.getOutputName();
+        String script = coverageView.getDefinition();
+        List<CoverageBand> outputBands = coverageView.getCoverageBands();
+        List<InputCoverageBand> inputBands = outputBands.get(0).getInputCoverageBands();
+        List<String> sourceBands =
+                inputBands.stream().map(icb -> icb.getCoverageName()).collect(Collectors.toList());
+
+        RenderedImage[] sources = new RenderedImage[inputCoverages.size()];
+        sources[0] = reference.getRenderedImage();
+        for (int i = 1; i < sources.length; i++) {
+            GridCoverage2D coverage = coverages[i];
+            double[] nodata = CoverageUtilities.getBackgroundValues(coverage);
+            ROI roi = CoverageUtilities.getROIProperty(coverage);
+            if (coverage.getGridGeometry().equals(reference.getGridGeometry())) {
+                sources[i] = coverage.getRenderedImage();
+            } else {
+                sources[i] = GridCoverage2DRIA.create(coverage, reference, nodata, GeoTools.getDefaultHints(), roi);
+            }
+        }
+
+        String[] sourceNames = sourceBands.toArray(new String[0]);
+        int outputBandCount = outputBands.size();
+        Range[] nodatas = new Range[sources.length];
+        nodatas = getNodatas(nodatas, coverages);
+        RenderedOp result = JiffleDescriptor.create(
+                sources,
+                sourceNames,
+                destName,
+                script,
+                null,
+                null,
+                outputBandCount,
+                null,
+                null,
+                nodatas,
+                GeoTools.getDefaultHints());
+
+        GridSampleDimension[] sampleDimensions = getSampleDimensions(result, destName);
+        GridCoverageFactory factory = new GridCoverageFactory(GeoTools.getDefaultHints());
+        return factory.create("jiffle", result, reference.getEnvelope(), sampleDimensions, coverages, null);
+    }
+
+    private GridSampleDimension[] getSampleDimensions(RenderedOp result, String destName) {
+        SampleModel sm = result.getSampleModel();
+        Stream<String> names = getBandNames(sm.getNumBands(), destName);
+        SampleDimensionType sourceType = TypeMap.getSampleDimensionType(sm, 0);
+        NumberRange<? extends Number> range = TypeMap.getRange(sourceType);
+        double[] nodata = null; // {Double.NaN};
```

Comment:
Wondering about the nodata, when viewing with jpeg output I get "two backgrounds", a white for the data calculated by Jiffle (NaN) and a black outside. There might be an inconsistency at play:

![image](https://github.com/user-attachments/assets/8f0726c5-4e48-480d-8d9a-d9b030fade16)

Reply:
That's weird. I only get a black background when requesting untiled JPEG output.
For other cases including tiled
![image](https://github.com/user-attachments/assets/63b76cbc-e075-4682-bf9f-9758489f4353)

Untiled PNG.
![image](https://github.com/user-attachments/assets/28dc1976-8be7-4316-b125-0cda7c32ce54)

and default raster style 
![image](https://github.com/user-attachments/assets/3949c1b0-edeb-406b-9834-a37b368869f4)

I see good outputs.

Even from RenderedImageBrowser I can see good result.
![image](https://github.com/user-attachments/assets/dfd1998a-f949-4c31-9ac2-886f031c1bf4)

I'm wondering if that could be a specific issue with the JPEG encoding.